google/cloud/aiplatform/jobs.py

```diff
@@ -1913,3 +1918,522 @@ def run(
+                which stores logs. A day is the basic unit of
+                the TTL and we take the ceil of TTL/86400(a
+                day). e.g. { second: 3600} indicates ttl = 1
+                day.
+
+            stats_anomalies_base_directory (str):
+                Optional. Stats anomalies base folder path.
+
+            enable_monitoring_pipeline_logs (bool):
+                Optional. If true, the scheduled monitoring pipeline logs are sent to
+                Google Cloud Logging, including pipeline status and
+                anomalies detected. Please note the logs incur cost, which
+                are subject to `Cloud Logging
+                pricing <https://cloud.google.com/logging#pricing>`__.
+
+            labels (Dict[str, str]):
+                Optional. The labels with user-defined metadata to
+                organize the ModelDeploymentMonitoringJob.
+                Label keys and values can be no longer than 64
+                characters (Unicode codepoints), can only
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed. See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+
+            encryption_spec_key_name (str):
+                Optional. Customer-managed encryption key spec for a
+                ModelDeploymentMonitoringJob. If set, this
+                ModelDeploymentMonitoringJob and all
+                sub-resources of this
+                ModelDeploymentMonitoringJob will be secured by
+                this key.
+
+            create_request_timeout (int):
+                Optional. Timeout in seconds for the model monitoring job creation request
+
+        Returns:
+            An instance of ModelDeploymentMonitoringJob
+        """
+        if not display_name:
+            display_name = cls._generate_display_name()
+
+        utils.validate_display_name(display_name)
+
+        if labels:
+            utils.validate_labels(labels)
+
+        if stats_anomalies_base_directory:
+            stats_anomalies_base_directory = gca_io_compat.GcsDestination(
+                output_uri_prefix=stats_anomalies_base_directory
+            )
+
+        if encryption_spec_key_name:
+            encryption_spec_key_name = gca_encryption_spec_compat.EncryptionSpec(
+                kms_key_name=encryption_spec_key_name
+            )
+
+        self = cls._empty_constructor(
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        parent = initializer.global_config.common_location_path(
+            project=self.project,
+            location=self.location,
+        )
+
+        api_client = self.api_client
+
+        _LOGGER.log_create_with_lro(cls)
+
+        if isinstance(endpoint, str):
+            endpoint = aiplatform.Endpoint(endpoint).resource_name
```

Comment:
The credentials that were used instantiate this `Endpoint` are lost here. It may be better to pass `aiplatform.Endpoint` instead of `str` to `_parse_configs`.